devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,182 @@
+
+		// Samsung LED
+		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0300", outClusters: "0019", manufacturer: "Samsung Electronics", model: "SAMSUNG-ITM-Z-003", deviceJoinName: "Samsung Light", mnmn: "Samsung Electronics", vid: "SAMSUNG-ITM-Z-003"
+	}
+
+	// UI tile definitions
+	tiles(scale: 2) {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label: '${name}', action: "off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "off", label: '${name}', action: "on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "turningOn", label: '${name}', action: "off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel"])
+	}
+}
+
+private getFAN_CLUSTER_VALUE() { 0x0202 }
+private getFAN_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+
+def parse(String description) {
+	// Parse incoming device messages to generate events
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)	
+	if (event) {
+		sendEvent(event)
+	} else if (description?.startsWith('read attr -')) {
+		def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+		if (zigbeeMap.clusterInt == FAN_CLUSTER_VALUE &&
+		    zigbeeMap.attrInt == FAN_STATUS_VALUE) {		
+			log.debug "read attribute event for fan cluster attrib FAN_STATUS"
+			def childDevice = childDevices.find {
+				//find light child device
+				log.debug "parse() child device found"
+				it.device.deviceNetworkId == "${device.deviceNetworkId}:1" 
+			}          
+			event.displayed = true
+			zigbeeMap.name = "fanSpeed"
+			childDevice.sendEvent(zigbeeMap)
+			if (zigbeeMap.value == "00") {
+				log.debug "fan_off => switch off"
+				childDevice.sendEvent(name: "switch", value: "off")
+				childDevice.sendEvent(name: "level", value: 0)	// For cloud to cloud device UI update
+			} else {
+				log.debug "fan_on => switch on"
+				childDevice.sendEvent(name: "switch", value: "on")
+				def int_v  = zigbeeMap.value as Integer
+				int_v = int_v * 25
+				int_v = int_v > 100 ? 100 : int_v
+				log.debug "child device Level set $int_v"
+				childDevice.sendEvent(name: "level", value: int_v)	// For cloud to cloud device UI update
```

Comment:
Yeah, I get it. I clearly meant that you perhaps should send a `fanSpeed` event in addition, not instead.

Reply:
Because of what I was addressing here https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/68445/files#r686664824 I believe there is no correct `fanSpeed` event send as a response of Zigbee command.

Reply:
The line 79 zigbeeMap.name = "fanSpeed" means designate a child UI event name. And the line 89 to 90 means matching the fan speed to switch level.

Reply:
Even though it's working, cause `zigbeeMap` indeed contains key `value`, I'd like you to create a separate map object, like `fanSpeedEvent` and send it instead. This way it'll be more clear what are you doing.

Reply:
Oh, great! Not for cause the confusion I'll change the name.